### PR TITLE
Support package level annotations

### PR DESCRIPTION
**Pull request by ndeloof**
see https://github.com/jenkinsci/configuration-as-code-plugin/issues/278 for initial discussion

see https://github.com/kohsuke/access-modifier/pull/13 for actual usage

### CODE REVIEW

**Comment by ndeloof** [comment]
made a minor change as this makes access-modifier easier to implement:
package-info annotations are indexed by package name (nor listing classes in package)
drawback doing so is that such entries in index will produce `warning: class file not found` when an older version of access-modifier is used.

**Comment by ndeloof** [comment]
@oleg-nenashev can you merge (and release) this or do you expect more reviews ?

**Comment by oleg-nenashev** [comment]
@jglick as the last person to release it, could you please also take a look?

File: src/main/java/org/jvnet/hudson/annotation_indexer/AnnotationProcessorImpl.java [hunk 1]
```diff
@@ -80,6 +81,10 @@ void add(Element elt) {
             case CONSTRUCTOR:
                 t = (TypeElement) elt.getEnclosingElement();
                 break;
+            case PACKAGE:
+                classes.add(((PackageElement)elt).getQualifiedName().toString());
```
Comment (jglick) [inline src/main/java/org/jvnet/hudson/annotation_indexer/AnnotationProcessorImpl.java]: This is not a class name. That will break [here](https://github.com/jenkinsci/lib-annotation-indexer/blob/1da4d7c9a800fd1beb61895355e19f421a4c4ab3/src/main/java/org/jvnet/hudson/annotation_indexer/Index.java#L99-L107).

Comment (ndeloof) [reply src/main/java/org/jvnet/hudson/annotation_indexer/AnnotationProcessorImpl.java]: indeed. And java reflexion don't let us list classes in a package using reflexion :'(
What do you think about the [initial commit](https://github.com/jenkinsci/lib-annotation-indexer/pull/5/commits/f913f1a43a8668d1fca3f795df0fdd70d043f7f8) to just add all classes within package to the index ? this makes the accmod implementation less efficient, but would reduce impact ?

Comment (ndeloof) [reply src/main/java/org/jvnet/hudson/annotation_indexer/AnnotationProcessorImpl.java]: updated by https://github.com/jenkinsci/lib-annotation-indexer/pull/5/commits/1952a83f42722e94ad4e7d5dcc993f1d6331628e to index com.foo.* and list Package as indexed element

https://github.com/kohsuke/access-modifier/pull/13 updated accrodingly

**Comment by ndeloof** [comment]
@jglick can you please reconsider your review with latest changes ?

**Comment by ndeloof** [comment]
@jglick ping

**Comment by ndeloof** [comment]
inactive, feel free to take over

**Comment by oleg-nenashev** [comment]
I will take it over, thanks!

**Comment by ndeloof** [comment]
@oleg-nenashev could you please cherry-pick those commit into your own fork so I don't get notified on this PR, I also plan to cleanup my hundreds jenkins-* forks
thanks

**Comment by oleg-nenashev** [comment]
Please give me few days to do so

**Comment by ndeloof** [comment]
@oleg-nenashev did you pulled my commits ?

**Comment by oleg-nenashev** [comment]
I just got it merged. Should be good enough for now, thanks!

**Pull request merged**